sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -22,8 +28,23 @@ impl BearerTokenCredentialPolicy {
         Self {
             credential,
             scopes: scopes.into_iter().map(|s| s.into()).collect(),
+            access_token: None,
+            last_refresh_time: OffsetDateTime::now_utc().unix_timestamp(),
         }
     }
+
+    fn scopes(&self) -> Vec<&str> {
+        self.scopes
+            .iter()
+            .map(String::as_str)
+            .collect::<Vec<&str>>()
+    }
+
+    async fn need_new_token(&mut self) {
+        self.last_refresh_time = OffsetDateTime::now_utc().unix_timestamp();
+        let access_token = self.credential.get_token(&self.scopes()).await;
+        self.access_token = Some(access_token.unwrap());
```

Comment:
Don't call `unwrap()` in production code unless you know for sure it'll never fail. This can.

1. This function is probably better as `refresh()`.
2. This function should return an `azure_core::Result<()>`.
3. Then on line 45, just do `... .await?;`